Build failure: linuxPackages_6_13.rtl8821ce

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

```
nix build github:nixos/nixpkgs#linuxPackages_6_13.rtl8821ce
```

Or if that doesn't work try with `nix repl`:
```
nix-repl> :lf github:nixos/nixpkgs
Added 16 variables.

nix-repl> :a outputs.legacyPackages.x86_64-linux
Added 23571 variables.

nix-repl> :b linuxPackages_6_13.rtl8821ce
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/286077333

### Relevant log output

```console
make[1]: Entering directory '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/build'
/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/Makefile:251: update target '__sub-make' due to: target is .PHONY
make --no-print-directory -C /nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/build \
-f /nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/Makefile modules
Putting child 0x46c8b0 (__sub-make) PID 2729003 on the chain.
 Recipe of '__sub-make' is being run.
The prerequisites of 'modules' are being made.
Live child 0x46c8b0 (__sub-make) PID 2729003
GNU Make 4.4.1
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Reading makefiles...
Reading makefile '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/Makefile'...
Reading makefile '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/scripts/Kbuild.include' (search path) (no ~ expansion)...
Reading makefile '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/scripts/subarch.include' (search path) (no ~ expansion)...
file: Failed to open '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/config/kernel.release': No such file or directory
file: Failed to open '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/config/kernel.release': No such file or directory
file: Failed to open '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/config/kernel.release': No such file or directory
file: Failed to open '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/config/kernel.release': No such file or directory
file: Failed to open '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/config/kernel.release': No such file or directory
Reading makefile '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/scripts/Makefile.compiler' (search path) (no ~ expansion)...
Reading makefile '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/config/auto.conf' (search path) (no ~ expansion)...
Reading makefile '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/arch/x86/Makefile' (search path) (no ~ expansion)...
/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/scripts/Makefile.compiler:51: not recursively expanding cc-option to export to shell function
file: Failed to open '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/config/kernel.release': No such file or directory
mkdir: cannot create directory ‘.tmp_2729018’: Read-only file system
/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/scripts/Makefile.compiler:51: not recursively expanding cc-option to export to shell function
file: Failed to open '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/config/kernel.release': No such file or directory
mkdir: cannot create directory ‘.tmp_2729021’: Read-only file system
/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/scripts/Makefile.compiler:51: not recursively expanding cc-option to export to shell function
file: Failed to open '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/config/kernel.release': No such file or directory
mkdir: cannot create directory ‘.tmp_2729024’: Read-only file system
```

### Additional context

Contrast this with `linuxPackages_6_12.rtl8821ce` (which builds fine):
```
make[1]: Entering directory '/nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/build'
/nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/source/Makefile:224: update target '__sub-make' due to: target is .PHONY
make --no-print-directory -C /nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/build \
-f /nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/source/Makefile modules
Putting child 0x471b00 (__sub-make) PID 2732080 on the chain.
 Recipe of '__sub-make' is being run.
The prerequisites of 'modules' are being made.
Live child 0x471b00 (__sub-make) PID 2732080
GNU Make 4.4.1
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Reading makefiles...
Reading makefile '/nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/source/Makefile'...
Reading makefile '/nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/source/scripts/Kbuild.include' (search path) (no ~ expansion)...
Reading makefile '/nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/source/scripts/subarch.include' (search path) (no ~ expansion)...
Reading makefile '/nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/source/scripts/Makefile.compiler' (search path) (no ~ expansion)...
Reading makefile 'include/config/auto.conf' (search path) (no ~ expansion)...
Reading makefile '/nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/source/arch/x86/Makefile' (search path) (no ~ expansion)...
Reading makefile '/nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/source/scripts/Makefile.extrawarn' (search path) (no ~ expansion)...
Reading makefile '/nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/source/scripts/Makefile.debug' (search path) (no ~ expansion)...
Reading makefile '/nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/source/scripts/Makefile.btf' (search path) (no ~ expansion)...
Reading makefile '/nix/store/vrchq86cyw1v0vknns6alh4gkbh6wb0n-linux-6.12.9-dev/lib/modules/6.12.9/source/scripts/Makefile.gcc-plugins' (search path) (no ~ expansion)...
Reading makefile 'arch/x86/include/generated/asm/.orc_hash.h.cmd' (search path) (don't care) (no ~ expansion)...
Using jobserver controller fifo:/tmp/nix-shell-2374622-0/nix-shell.wF7YSf/GMfifo2732059
Updating makefiles....
Makefile 'include/config/auto.conf' might loop; not remaking it.
```

For some reason `nix build` doesn't seem to even try anymore:
```
$ nix build -L github:nixos/nixpkgs#linuxPackages_6_13.rtl8821ce
error: some outputs of '/nix/store/64hcyld673d4wwmcxm9p3d9q4m2ia7r0-rtl8821ce-6.13-unstable-2024-03-26.drv' are not valid, so checking is not possible
```
But `nix repl` can still reproduce this:
```
nix-repl> :lf github:nixos/nixpkgs
Added 16 variables.

nix-repl> :a outputs.legacyPackages.x86_64-linux
Added 23571 variables.

nix-repl> :b linuxPackages_6_13.rtl8821ce
error: builder for '/nix/store/64hcyld673d4wwmcxm9p3d9q4m2ia7r0-rtl8821ce-6.13-unstable-2024-03-26.drv' failed with exit code 2;
       last 25 log lines:
       > mkdir: cannot create directory '.tmp_98': Permission denied
       > mkdir: cannot create directory '.tmp_101': Permission denied
```

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.9, NixOS, 25.05 (Warbler), 25.05.20250120.7ceed4f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/x46a0a227r8y8bqvld0d4hhxazlwsf3x-source`
 
(on 6.12.9; can't upgrade due to this build failure)

### Notify maintainers

@hhm0 @Defelo 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

It fixed that error but throws a different one now:
```
$ nix build -L "github:nixos/nixpkgs?ref=master#linuxPackages_6_13.rtl8821ce"
---------------stripped----------
rtl8821ce> os_dep/linux/ioctl_cfg80211.c:9674:32: error: initialization of 'int (*)(struct wiphy *, struct net_device *, struct cfg80211_chan_def *)' from incompatible pointer type 'int (*)(struct wiphy *, struct cfg80211_chan_def *)' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
rtl8821ce>  9674 |         .set_monitor_channel = cfg80211_rtw_set_monitor_channel,
rtl8821ce>       |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
---------------stripped----------
```

--- a/pkgs/os-specific/linux/rtl8821ce/default.nix
+++ b/pkgs/os-specific/linux/rtl8821ce/default.nix
@@ -41,10 +41,7 @@ stdenv.mkDerivation (finalAttrs: {
     homepage = "https://github.com/tomaspinho/rtl8821ce";
     license = lib.licenses.gpl2Only;
     platforms = lib.platforms.linux;
-    maintainers = with lib.maintainers; [
-      hhm
-      defelo
-    ];
+    maintainers = with lib.maintainers; [ defelo ];
     broken =
       stdenv.hostPlatform.isAarch64
       || ((lib.versions.majorMinor kernel.version) == "5.4" && kernel.isHardened);

--- a/pkgs/os-specific/linux/rtl8821ce/default.nix
+++ b/pkgs/os-specific/linux/rtl8821ce/default.nix
@@ -25,9 +25,9 @@ stdenv.mkDerivation (finalAttrs: {
 
   prePatch = ''
     substituteInPlace ./Makefile \
-      --replace /lib/modules/ "${kernel.dev}/lib/modules/" \
-      --replace /sbin/depmod \# \
-      --replace '$(MODDESTDIR)' "$out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/"
+      --replace-fail /lib/modules/ "${kernel.dev}/lib/modules/" \
+      --replace-fail /sbin/depmod \# \
+      --replace-fail '$(MODDESTDIR)' "$out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/"
   '';
 
   preInstall = ''

--- a/pkgs/os-specific/linux/rtl8821ce/default.nix
+++ b/pkgs/os-specific/linux/rtl8821ce/default.nix
@@ -9,13 +9,13 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "rtl8821ce";
-  version = "${kernel.version}-unstable-2024-03-26";
+  version = "${kernel.version}-unstable-2025-02-08";
 
   src = fetchFromGitHub {
     owner = "tomaspinho";
     repo = "rtl8821ce";
-    rev = "f119398d868b1a3395f40c1df2e08b57b2c882cd";
-    hash = "sha256-EfpKa5ZRBVM5T8EVim3cVX1PP1UM9CyG6tN5Br8zYww=";
+    rev = "46d1a59e37364ed4aedcb09746f0ae412e6b2066";
+    hash = "sha256-vQM7e8tuy4Lf2HvGtHUvOxulF4DIhX33vFel9u9i178=";
   };
 
   hardeningDisable = [ "pic" ];

